### DISCUSSION TITLE
How to properly upgrade neuron via Nix?

### DISCUSSION THREAD

**Discussion by ouuan** [srid/neuron#643] [Installation]
In the [documentation](https://neuron.zettel.page/install#install-neuron), it says "this command can also *upgrade* your existing install of neuron". But when I run `nix-env -if https://github.com/srid/neuron/archive/master.tar.gz`, it takes the same time as the first install, and Nix uses another 6GiB of my disk space.

So, what's the proper way to upgarde neuron, so that it takes less time than the first install, or at least doesn't take more disk space?

Here's what my `/nix/store` looks like:

![du](https://user-images.githubusercontent.com/30581822/127762701-ca1bd03a-8a36-4367-98ba-dd9e7ec29dca.png)

**Comment by ouuan** [comment]
BTW, I discovered that the "Enable cache" step can save a lot of space (6GiB -> 1GiB) and time just now. The "(optional)" and "spend some time compiling" in the documentation are technically correct, but it made me incorrectly think that the cache is not very useful (as an Arch Linux user, I don't like unnecessary steps/optional stuff). It could be better if the "optional" could be replaced by "recommended", and "spend some time compiling" could be replaced by "using the cache can save you ~5GiB disk space and several minutes to compile".

**Reply by srid** [reply]
That's good to know; we should then add it to the install docs.

**Comment by srid** [comment]
`sudo nix-collect-garbage --delete-older-than 3d` (adjust the days accordingly) will clean up unused derivations in /nix.
